### PR TITLE
UML-2690: Adding code created by actor id to viewer code table to be used when lpa is removed from account

### PR DESCRIPTION
**Pull request by SeemaMenon**
# Purpose

https://opgtransform.atlassian.net/browse/UML-2690

Fixes UML-2690

## Approach

_Explain how your code addresses the purpose of the change_

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Use a Lasting Power of Attorney service_

## Checklist

* [X] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] New event_codes have been documented on the [wiki page](https://opgtransform.atlassian.net/wiki/spaces/LSML2/pages/3277881441/Understanding+the+event+logs)
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [ ] I have added welsh translation tags and updated translation files
* [ ] I have run an accessibility tool on any pages I have made changes to and fixed any issues found
* [ ] I have notified the Interaction Designer of any content changes so that appropriate screenshots/flow diagram changes can be made
* [ ] The product team have tested these changes

### CODE REVIEW

File: service-api/app/test/AppTest/DataAccess/DynamoDb/ViewerCodesTest.php [hunk 1]
```diff
@@ -85,10 +90,13 @@ public function cannot_lookup_a_missing_code(): void
...
-            }))
-            ->willReturn($this->createAWSResult([
-                'Item' => []
-            ]));
+            })
+        )
+            ->willReturn(
+                $this->createAWSResult([
+                                           'Item' => [],
+                                       ])
```
Comment (cooperaj) [inline service-api/app/test/AppTest/DataAccess/DynamoDb/ViewerCodesTest.php:98]:
```suggestion
                $this->createAWSResult(
                    [
                        'Item' => [],
                    ]
                )
```
Something like this.

**Review by cooperaj — CHANGES_REQUESTED** [review]
Looks like a code cleanup tool or something was run but it's not used our linting rules so it's changed a bunch of stuff it shouldn't

File: service-api/app/src/App/src/Handler/LpasResourceHandler.php [hunk 2]
```diff
@@ -24,7 +24,7 @@ class LpasResourceHandler implements RequestHandlerInterface
 {
     public function __construct(
         private LpaService $lpaService,
-        private RemoveLpa $removeLpa,
+        private RemoveLpa $removeLpa
```
[the remark this answers is not in the document]
Comment (SeemaMenon) [reply service-api/app/src/App/src/Handler/LpasResourceHandler.php:27]: I am not sure, looks like something has changes my settings. Corrected manually for now.

File: service-api/app/src/App/src/Service/Lpa/RemoveLpa.php [hunk 3]
```diff
@@ -69,13 +69,6 @@ public function __invoke(string $userId, string $token): array
         if (!empty($viewerCodes)) {
             foreach ($viewerCodes as $viewerCodeRecord) {
                 $this->viewerCodesRepository->removeActorAssociation($viewerCodeRecord['ViewerCode']);
-                if (
```
[the remark this answers is not in the document]
Comment (SeemaMenon) [reply service-api/app/src/App/src/Service/Lpa/RemoveLpa.php:72]: Business requirement changed to now NOT cancel codes when a LPA is deleted from account. The generated access code should remain active [until it expires] for users to view and the same will be shown in active code section in 'Check Access Code' page.

File: service-api/app/features/context/Acceptance/LpaContext.php [hunk 4]
```diff
@@ -2335,20 +2346,7 @@ public function iClickToCheckMyAccessCodesThatIsUsedToViewLPA(): void
...
-                            ]
-                        ),
-                        $this->marshalAwsResultData(
-                            [
-                                'Viewed' => '2020-10-01T23:59:59+00:00',
-                                'ViewerCode' => $this->accessCode,
-                                'ViewedBy' => 'organisation2'
-                            ]
-                        )
+                        $this->marshalAwsResultData($code1)
```
[the remark this answers is not in the document]
Comment (SeemaMenon) [reply service-api/app/features/context/Acceptance/LpaContext.php]: No, I shouldn't have. Added back in .

**Review by cooperaj — APPROVED** [review]

**Pull request merged**